get_maintainer.py

```diff
@@ -0,0 +1,532 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Lists maintainers for files or commits. Similar in function to
+scripts/get_maintainer.pl from Linux, but geared towards GitHub. The mapping is
+in MAINTAINERS.yml.
+
+The comment at the top of MAINTAINERS.yml in Zephyr documents the file format.
+
+See the help texts for the various subcommands for more information. They can
+be viewed with e.g.
+
+    ./get_maintainer.py path --help
```

Comment:
Hmm, actually, shouldn't we make this a west extension? That way it could be invoked as:

`west get-maintainer path`

or so. Here is the [extension documentation](https://docs.zephyrproject.org/latest/guides/west/extensions.html).

No worries if you want an independently runnable script which is also a library also, that makes sense, but I think since this is user facing, the primary user interface should be west.

Reply:
You're free to add a west extension. I think we should move away from the big monolithic tool approach and have simple and easy to debug standalone script instead, so I'd rather not.

Reply:
> simple and easy to debug standalone scripts instead

Why are they mutually exclusive in this case?